common.sh

```diff
@@ -187,9 +187,35 @@ create_instance()
 get_instance_ip()
 {
     execution_seq=$((${execution_seq}+1))
-    INSTANCE_IPS=$(aws ec2 describe-instances --instance-ids ${INSTANCE_IDS[@]} \
-                        --query "Reservations[*].Instances[*].PrivateIpAddress" \
-                        --output=text)
+    local retry=20
+    local ret=0
+    local bash_option=$-
+    local restore_e=0
+    if [[ $bash_option =~ e ]]; then
+        restore_e=1
+        set +e
+    fi
+    while [ $retry -ge 0 ]; do
+        INSTANCE_IPS=$(aws ec2 describe-instances --instance-ids ${INSTANCE_IDS[@]} \
+                            --query "Reservations[*].Instances[*].PrivateIpAddress" \
+                            --output=text)
+        ret=$?
+        if [ "$ret" -eq 0 ] && [ -n "$INSTANCE_IPS" ]; then
```

Comment:
Should we also check the number of returned IPs here to make sure it's equal to the number of the given instance_ids?

Reply:
yes; good catch.  There's no guarantee of all or nothing here.

Reply:
Added this check in latest revision, thanks.